docs/source/backup/index.rst

```diff
@@ -1,272 +1,255 @@
-    If none of these variables are set (``end_date_of_backup`` and ``days_to_backup``), then the end date of backup is set to the current date.
-  | E.g. ``"end_date_of_backup": null`` or ``"end_date_of_backup": "2015-07-20 11:13:08.145804+02:00"``.
+.. _restore_postgresql:
 
-* | ``days_to_backup`` (in `days` or ``null``):
-  | If set, you specify how many days you will backup from the starting date of your backup.
-    If it is set to ``null`` and also ``end_date_of_backup`` is set to ``null``, then the end date of the backup is set to the current date.
-    You can not set ``days_to_backup`` & ``end_date_of_backup`` at the same time (it will lead to an error).
-  | E.g. ``"days_to_backup": null`` or ``"days_to_backup": 5``.
+Restore backup
+++++++++++++++
 
-* | ``backup_length_threshold`` (in `hours`):
-  | The backup script runs in rounds and on every round it will backup a number of days that are controlled primarily by ``periodicity`` and also by ``end_date_of_backup`` / ``days_to_backup``, for the last backup round.
-    The ``backup_length_threshold`` specifies the *lowest acceptable* round length.
-    This is important for the end of the backup.
-  | E.g. ``backup_length_threshold: 1``
+In order to restore a backup, you will need to:
 
-* | ``backup_dir`` (absolute path):
-  | The destination directory of the backup.
-  | E.g. ``"backup_dir": "/home/USERNAME/.aiida/backup/backup_dest"``.
+ 1. | Restore the repository folder.
+    | If you used the :ref:`backup script <backup_script>`, this involves extracting the corresponding zip file (should be created in ``~/.aiida/``).
 
-To start the backup, run the ``start_backup.py`` script.
-Run as often as needed to complete a full backup, and then run it periodically (e.g. calling it from a cron script, for instance every day) to backup new changes.
+ 2. | Create an empty database following the instructions described in :ref:`database` skipping the ``verdi setup`` phase.
+    | The database should have the same name and database username as the original one (i.e. if you are restoring on the original postgresql cluster, you may have to either rename or delete the original database).
 
-.. note::
+ 3. As the ``postgres`` user, ``cd`` to the folder containing the database dump (``aiidadb-backup.psql``) and load it::
 
-  You can set up a cron job using the following command::
+      psql -h localhost -U aiida -d aiidadb -f aiidadb-backup.psql
 
-    sudo crontab -u USERNAME -e
 
-  It will open an editor, where you can add a line of the form::
+    After supplying your database password, the database should be restored.
 
-    00 03 * * * /home/USERNAME/.aiida/backup/start_backup.py 2>&1 | mail -s "Incremental backup of the repository" USER_EMAIL@domain.net
+    .. note::
 
-  or (if you need to backup a different profile than the default one)::
+        On Ubuntu, type ``sudo su - postgres`` to become the postgres UNIX user.
```

Comment:
This should probably be shifted upwards to come before the first time one needs to do this.

Reply:
Ah yes - that would be in the section on "moving the database".

In my opinion, the "moving the database" section should either be moved out to some "recipe collection section" or removed entirely:
 * it is not really about backup
 * it contradicts what we say in the backup section (that you should not use the postgres files themselves)
 * it is only about postgres, no AiiDA specifics (`moving postgres data directory` gives lots of results on Google)
 * the same effect could usually be achieved with a proper backup + restore

I guess it can save you a bit of time if you indeed need to move the database, so we could move it to "Troubleshooting" under "Working with AiiDA" and link to it. Or just get rid of it.

@giovannipizzi What do you say?

Reply:
I think we should keep the section on moving the database, but where it's now probably is not useful to users.

The goal, to me, would be to create a section on "how to configure your computer for a performant use of AiiDA" and we give there all the tricks, including things like:
- put the DB in a fast disk, ideally on a big partition (and we link to the section on 'moving the db' to say that, if you already have the db, you can move it to a different partition following those instructions)
- perform incremental backups (and link to section)
- increase the number of workers
- anything else that we do in our production DBs but it's not really documented, or documented in a scattered way (e.g. remove Ubuntu's indexing of the folders etc.)

Feel free to open an issue with my message above if you don't want to address it here.


@ltalirz Note that I'm not sure that the same effect can be obtained with dump+restore - it's not just about moving a single DB, but the whole folder where Postgres stores *all* DBs.

Reply:
The "performant AiiDA usage" section sounds like a good idea to me.
It is actually already pretty much exists where I suggested to move things (Working with AiiDA => Troubleshooting).

I've added it now.
@sphuber : Would you like to add some more wisdom concerning the number of workers?
E.g. how to roughly estimate it and what number not to exceed (#cores? 4x#cores?)

Reply:
I've also added a "tips&tricks" section, starting from the one that was already there.